File: bestia-models/src/test/java/net/bestia/model/map/TilePropertiesTest.java
package net.bestia.model.map;

import org.junit.Assert;
import org.junit.Test;

public class TilePropertiesTest {
	
	@Test(expected=IllegalArgumentException.class)
	public void negWalkspeedSpeed_throws() {
		new TileProperties(true, -100);
	}
	
	@Test
	public void correctArgs_getterOk() {
		TileProperties p = new TileProperties(true, 100);
		Assert.assertTrue(p.isWalkable());
		Assert.assertEquals(100, p.getWalkspeed());
	}

}

File: bestia-models/src/test/kotlin/net/bestia/model/map/TilePropertiesTest.kt
package net.bestia.model.map

import org.junit.Assert
import org.junit.Test

class TilePropertiesTest {

  @Test(expected = IllegalArgumentException::class)
  fun negWalkspeedSpeed_throws() {
    TileProperties(true, -100, false)
  }

  @Test
  fun correctArgs_getterOk() {
    val (isWalkable, walkspeed) = TileProperties(true, 100, false)
    Assert.assertTrue(isWalkable)
    Assert.assertEquals(100, walkspeed.toLong())
  }
}
